BridgeAppSDK/SBAGenericStepDataSource.swift

```diff
@@ -0,0 +1,614 @@
+        }
+        
+        // if we selected an item and this is a single-selection group, then we iterate
+        // our other items and de-select them
+        
+        if selected && singleSelection {
+            items.forEach({
+                if $0 != items[index] { $0.selected = false }
+            })
+        }
+    }
+    
+    fileprivate func internalAnswer() -> AnyObject {
+        
+        if items.count > 0 {
+            
+            let answerFormat = items[0].formItem?.answerFormat
+            switch answerFormat {
+            case is ORKBooleanAnswerFormat:
+                return answerForBoolean()
+                
+            case is ORKMultipleValuePickerAnswerFormat,
+                 is ORKTextChoiceAnswerFormat:
+                return answerForTextChoice()
+                
+            default:
+                return _answer ?? defaultAnswer
+            }
+        }
+        return _answer ?? defaultAnswer
+    }
+    
+    fileprivate func setInternalAnswer(_ answer: AnyObject) {
+        
+        if items.count > 0 {
+            
+            let answerFormat = items[0].formItem?.answerFormat
+            switch answerFormat {
+            case is ORKBooleanAnswerFormat:
+                
+                // iterate our items and find the item with a value equal to our answer,
+                // then select that item
+                
+                let formattedAnswer = answer as? NSNumber
+                for item in items {
+                    if (item.choice?.value as? NSNumber)?.boolValue == formattedAnswer?.boolValue {
+                        item.selected = true
+                    }
+                }
+                
+            case is ORKTextChoiceAnswerFormat:
+                
+                // iterate our items and find the items with a value that is contained in our
+                // answer, which should be an array, then select those items
+                
+                if let arrayAnswer = answer as? Array<AnyObject> {
+                    for item in items {
+                        for selectedValue in arrayAnswer {
+                            if let choiceString = item.choice?.value as? String, let selectedString = selectedValue as? String {
+                                item.selected = choiceString == selectedString
+                            }
+                            else if let choiceNumber = item.choice?.value as? NSNumber, let selectedNumber = selectedValue as? NSNumber {
+                                item.selected = choiceNumber.intValue == selectedNumber.intValue
+                            }
+                        }
+                    }
+                }
+                
+            case is ORKMultipleValuePickerAnswerFormat:
+                
+                // TODO: Josh Bruhin, 6/12/17 - implement this answer format.
+                fatalError("setInternalAnswer for ORKMultipleValuePickerAnswerFormat not implemented")
```

Comment:
Sorry, this is used in some apps. You need to have a "not supported" case that can *not* use this step view controller for the cases where the UI doesn't support them and then be able to gracefully change to using the default UI from ResearchKit.